src/Simulation/QSharpFoundation/Convert/Convert.qs

```diff
@@ -164,7 +164,22 @@ namespace Microsoft.Quantum.Convert {
     /// the mappings of each Pauli operator in big-endian order
     /// `bits(Pn) ... bits(P0)`.
     function PauliArrayAsInt(paulis : Pauli[]) : Int {
-        body intrinsic;
+        let len = Length(paulis);
+        if len > 31 {
+            fail $"Cannot pack bits of Pauli array longer than 31 (got {len}).";
+        }
+
+        mutable result = 0;
+        for index in (len-1)..-1..0 {
+            let p = paulis[index];
+            set result <<<= 2;
+            if   p == PauliI { set result += 0; }
+            elif p == PauliX { set result += 1; }
+            elif p == PauliY { set result += 3; }
+            elif p == PauliZ { set result += 2; }
+            else { fail $"Unexpected Pauli value {p}."; }
```

Comment:
Not relevant to the PR directly, but by way of collecting feedback on language proposals, this looks like somewhere that https://github.com/microsoft/qsharp-language/issues/81 could be quite useful, perhaps?

Reply:
I added [this comment](https://github.com/microsoft/QuantumLibraries/issues/419#issuecomment-789346507) for myself.
I'll review [microsoft/qsharp-language#81](https://github.com/microsoft/qsharp-language/issues/81) in more details.